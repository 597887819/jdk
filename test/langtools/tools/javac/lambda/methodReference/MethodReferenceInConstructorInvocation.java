/*
 * @test /nodynamiccopyright/
 * @bug 8261006
 * @summary fail to parse broken interface::method in lambda
 * @compile/fail/ref=MethodReferenceInConstructorInvocation.out -XDrawDiagnostics MethodReferenceInConstructorInvocation.java
 */

import java.util.function.Supplier;

public class MethodReferenceInConstructorInvocation {
    interface Bar {
        default String getString() {
            return "";
        }
    }

    static class Foo implements Bar {

        public Foo() {
            this(Bar.super::getString);
        }

        public Foo(Supplier<String> sString) {
        }
    }
}
