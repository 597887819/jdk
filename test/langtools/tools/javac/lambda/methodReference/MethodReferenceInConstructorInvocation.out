MethodReferenceInConstructorInvocation.java:20:18: compiler.err.super.mref.cant.occur.static.context
1 error
